File: docs/guides/go/src/searchWithOptionalFilters.go
package main

import (
	"context"

	"github.com/algolia/algoliasearch-client-go/v4/algolia/next/search"
)

func reduceLabelsToFilters(_ []string) (search.OptionalFilters, error) {
	return search.OptionalFilters{}, nil // Implement your logic here
}

func searchWithOptionalFilters() {
	labels := []string{ /* A list of labels */ }

	client, err := search.NewClient("ALGOLIA_APPLICATION_ID", "ALGOLIA_API_KEY")
	if err != nil {
		// The client can fail to initialize if you pass an invalid parameter.
		panic(err)
	}

	optionalFilters, err := reduceLabelsToFilters(labels)
	if err != nil {
		panic(err)
	}

	searchParams := search.SearchParamsObjectAsSearchParams(
		search.NewSearchParamsObject().
			SetQuery("<YOUR_SEARCH_QUERY>").
			SetOptionalFilters(&optionalFilters),
	)

	_, err = client.SearchSingleIndex(context.Background(), "<YOUR_INDEX_NAME>", searchParams)
	if err != nil {
		panic(err)
	}
}

File: docs/guides/csharp/src/searchWithOptionalFilters.cs
namespace Algolia;

using System;
using System.Collections.Generic;
using System.Net.Http;
using System.Text.Json;
using Algolia.Search.Clients;
using Algolia.Search.Http;
using Algolia.Search.Models.Search;

class SearchWithOptionalFilters
{
  private static readonly List<string> labels =
  [ /* Your labels */
  ];

  private static OptionalFilters ReduceLabelsToFilters(List<string> labels)
  {
    return new OptionalFilters([]); // Implement your logic here
  }

  async Task Main(string[] args)
  {
    var client = new SearchClient(new SearchConfig("ALGOLIA_APPLICATION_ID", "ALGOLIA_API_KEY"));
    var optionalFilters = ReduceLabelsToFilters(labels);
    var searchParams = new SearchParams(
      new SearchParamsObject { Query = "<YOUR_SEARCH_QUERY>", OptionalFilters = optionalFilters }
    );

    await client.SearchSingleIndexAsync<Hit>("<YOUR_INDEX_NAME>", searchParams);
  }
}
